docs/network-policy.md

```diff
@@ -0,0 +1,361 @@
+# NetworkPolicy
+
+Kubernetes NetworkPolicy documentation: https://kubernetes.io/docs/concepts/services-networking/network-policies
+
+Kubernetes NetworkPolicy API reference: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#networkpolicy-v1-networking-k8s-io
+
+By default the network traffic from and to K8s pods is not restricted in any way. Using Network Policy is a way to enforce network isolation of selected pods. When a pod is selected by a NetworkPolicy allowed traffic is specified by the `ingress` and `egress` sections.  
+
+Each NetworkPolicy object consists of four sections:
+
+1. `podSelector`: a label selector that determines which pods the NetworkPolicy applies to
+2. `policyTypes`: determines which policy tpes are included, if none are selected then `Ingress` will always be set and `egress` will be set if any `egress` rules are applied 
```

Comment:
nit: should the capitalization be consistent? `Ingress` and `egress`

Reply:
Agree, I'm not the best at catching those things